pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/SimpleConsumerWrapper.java

```diff
@@ -385,7 +391,77 @@ public synchronized int getPartitionCount(String topic, long maxWaitTimeMs) {
         .addFetch(_topic, _partition, startOffset, 500000)
         .build());
 
-    return buildOffsetFilteringIterable(fetchResponse.messageSet(_topic, _partition), startOffset, endOffset);
+    if (!fetchResponse.hasError()) {
+      return buildOffsetFilteringIterable(fetchResponse.messageSet(_topic, _partition), startOffset, endOffset);
+    } else {
+      throw Errors.forCode(fetchResponse.errorCode(_topic, _partition)).exception();
+    }
+  }
+
+  /**
+   * Fetches the numeric Kafka offset for this partition for a symbolic name ("largest" or "smallest").
+   *
+   * @param requestedOffset Either "largest" or "smallest"
+   * @return An offset
+   */
+  public synchronized long fetchPartitionOffset(String requestedOffset) {
```

Comment:
please add a timeout argument. If needed, add a TODO here to add code to honor timeout (liek we have in other cases)